### PR TITLE
Clean up tests previously disabled for CC

### PR DESCRIPTION
**Pull request by mlopatkin**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test this

**Comment by mlopatkin** [comment]
@bot-gradle test this

File: platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/continuous/ChangesDuringBuildContinuousIntegrationTest.groovy [hunk 1]
```diff
@@ -29,29 +29,30 @@ class ChangesDuringBuildContinuousIntegrationTest extends AbstractContinuousInte
...
-            println "Modifying 'Thing.java' after initial compile task"
-            file("src/main/java/Thing.java").text = "class Thing { private static final boolean CHANGED=true; }"
-            file('change-triggered').text = 'done'
-        }
-    }
-    dependsOn 'classes'
-}
-jar.dependsOn postCompile
-"""
+        buildFile """
```
Comment (alllex) [inline platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/continuous/ChangesDuringBuildContinuousIntegrationTest.groovy:39]: 👍

File: subprojects/core/src/integTest/groovy/org/gradle/api/NestedConfigureDslIntegrationTest.groovy [hunk 2]
```diff
@@ -231,7 +229,7 @@ assert repositories.size() == 1
         errorOutput.contains("Could not find method maven() for arguments")
     }
 
-    @ToBeFixedForConfigurationCache(skip = INVESTIGATE)
+    @ToBeFixedForConfigurationCache(because = "resolves configuration at execution time through configuration container")
```
Comment (alllex) [inline subprojects/core/src/integTest/groovy/org/gradle/api/NestedConfigureDslIntegrationTest.groovy:232]: ❔  Can the test still be fixed for CC or is resolving the configuration at execution time inherent to the test?

**Review by alllex — APPROVED** [review]

File: subprojects/core/src/integTest/groovy/org/gradle/api/NestedConfigureDslIntegrationTest.groovy (hunk 2 above)
Comment (mlopatkin) [reply subprojects/core/src/integTest/groovy/org/gradle/api/NestedConfigureDslIntegrationTest.groovy:232]: > ❔ Can the test still be fixed for CC or is resolving the configuration at execution time inherent to the test?

I've tried, but it breaks output checking. The root cause is the same (Closure delegate not being available), just the message is slightly different.

**Pull request merged**